app/src/org/commcare/views/ViewUtil.java

```diff
@@ -123,4 +85,99 @@ public static int getColorDrawableColor(ColorDrawable drawable) {
             return drawable.getColor();
         }
     }
+
+
+    /**
+     * Determine width of each child view, based on mHints, the suite's size hints.
+     * mHints contains a width hint for each child view, each one of
+     * - A string like "50%", requesting the field take up 50% of the row
+     * - A string like "200", requesting the field take up 200 pixels
+     * - Null, not specifying a width for the field
+     * This function will parcel out requested widths and divide remaining space among unspecified columns.
+     *
+     * @param fullSize Width, in pixels, of the containing row.
+     * @return Array of integers, each corresponding to a child view,
+     * representing the desired width, in pixels, of that view.
+     */
+    public static int[] calculateColumnWidths(ArrayList<String> hints, int fullSize) {
+        // Convert any percentages to pixels. Percentage columns are treated
+        // as percentage of the entire screen width.
+        int[] widths = new int[hints.size()];
+        parseWidths(hints, widths, fullSize);
+
+        Pair<Integer, Integer> constraints = buildConstraints(widths);
+        int claimedSpace = constraints.first;
+        int indeterminateColumns = constraints.second;
+
+        if (claimedSpaceNotTight(fullSize, claimedSpace, indeterminateColumns)) {
+            // Either more space has been claimed than the screen has room for,
```

Comment:
would suggest making this comment a javadoc for the `claimedSpaceNotTight` method